### PR TITLE
Implement iam groups iam add-binding Subcommand

### PR DESCRIPTION
**Pull request by itsjaspermilan**
### Changes proposed in this PR:

Added the following `add-binding` subcommand to `hcp iam groups iam`

```
hcp iam groups iam add-binding \
    --group=iam/organization/cf8ef907-b9b9-4f2f-0000-e290448f0000/group/Group-Name \
    --member=ef938a22-09cf-4be9-0000-1f4587f80000 \
    --role=roles/iam.group-manager

hcp iam groups iam add-binding \
    --group=Group-Name \
    --member=ef938a22-09cf-4be9-0000-1f4587f80000 \
    --role=roles/iam.group-manager
```

Added some documentation on how to make a principal a group manager:
```
$ hcp iam groups iam
USAGE
  hcp iam groups iam <command> [Optional Flags]

DESCRIPTION
  The hcp iam groups iam command group lets you manage a group's IAM Policy.
  
  To set a member as a group manager, you can use the add-binding subcommand with
  the roles/iam.group-manager role.
  
  $ hcp iam groups iam add-binding \
    --group=8647ae06-ca65-467a-b72d-edba1f908fc8 \
    --member=ef938a22-09cf-4be9-b4d0-1f4587f80f53 \
    --role=roles/iam.group-manager

COMMANDS
  add-binding:  Add an IAM policy binding for a group.
  read-policy:  Read the IAM policy for a group.
```

### How I've tested this PR:

- Added a member to a group and set them as Group Manager
- Added the same binding twice and confirmed that it only set it once
- Tried to set the binding with a non-existing principal ID

### How I expect reviewers to test this PR:

- `make go/test`
- `make go/build && make go/install`
- Add a member to a group (can use UI or CLI)
- `hcp iam groups iam add-binding ...`
- `hcp iam groups iam read-policy ...`

### Output of affected commands:

```
$ hcp iam groups iam add-binding \
    -g=Cool-Group \
    -m=3be9dd66-bc21-4320-0000-5a7dad4c0000 \
    -r=roles/iam.group-manager
✓ Principal "3be9dd66-bc21-4320-0000-5a7dad4c0000" bound to role "roles/iam.group-manager".
```

### Checklist:
- [x] Tests added if applicable
- [x] CHANGELOG entry added or label 'pr/no-changelog' added to PR
  > Run `CHANGELOG_PR=<PR number> make changelog/new-entry` for guidance
  > in authoring a changelog entry, and commit the resulting file, which should
  > have a name matching your PR number. Entries should use imperative present
  > tense (e.g. Add support for...)

### CODE REVIEW

File: internal/commands/iam/groups/iam/add_binding.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		Name:      "add-binding",
+		ShortHelp: "Add an IAM policy binding for a group.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp iam groups iam add-binding" }}
+		command adds an IAM policy binding for the given group. A binding grants the
+		specified principal the given role on the group.
+
+		To view the available roles to bind, run {{ template "mdCodeOrBold" "hcp iam roles list" }}.
+
+		Currently, the only supported role on a principal in a group is {{ template "mdCodeOrBold" "roles/iam.group-manager" }}.
```
Comment (itsjaspermilan) [inline internal/commands/iam/groups/iam/add_binding.go:39]: Open to suggestions on this bit of documentation.

Comment (dadgar) [reply internal/commands/iam/groups/iam/add_binding.go:39]: I think this is good, maybe add a sentence after that says what a group manager can do

File: internal/commands/iam/groups/iam/add_binding.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: heredoc.New(ctx.IO).Must(`Bind a principal to role {{ template "mdCodeOrBold" "roles/iam.group-manager" }}:`),
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp iam groups iam add-binding \
+				  --group=8647ae06-ca65-467a-b72d-edba1f908fc8 \
```
Comment (dadgar) [inline internal/commands/iam/groups/iam/add_binding.go:46]: This is the group id? Should it be the name?

File: internal/commands/iam/groups/iam/add_binding.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "group",
+					Shorthand:    "g",
+					DisplayValue: "NAME",
+					Description:  "The name of the group to add the role binding to.",
+					Value:        flagvalue.Simple("", &opts.GroupName),
+					Required:     true,
```
Comment (dadgar) [inline internal/commands/iam/groups/iam/add_binding.go:60]: Could this be autocompleted?

File: internal/commands/iam/groups/iam/add_binding.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+	GroupName      string
+	PrincipalID    string
+	Role           string
+	ResourceClient resource_service.ClientService
+}
+
+func addBindingRun(opts *AddBindingOpts) error {
+	_, err := opts.Setter.AddBinding(opts.Ctx, opts.PrincipalID, opts.Role)
+	if err != nil {
+		fmt.Fprintf(opts.IO.Err(), "Principal %q bound to role %q.\n", opts.PrincipalID, opts.Role)
```
Comment (dadgar) [inline internal/commands/iam/groups/iam/add_binding.go:125]: Delete?

File: internal/commands/iam/groups/iam/iam.go [hunk 5]
```diff
@@ -12,14 +12,21 @@ func NewCmdIAM(ctx *cmd.Context) *cmd.Command {
...
-		manage a group's IAM Policy.
+		LongHelp: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+		The {{ template "mdCodeOrBold" "hcp iam groups iam" }} command group lets you manage a group's IAM Policy.
+
+		To set a member as a group manager, you can use the {{ template "mdCodeOrBold" "add-binding" }} subcommand with the {{ template "mdCodeOrBold" "roles/iam.group-manager" }} role.
+
+		$ hcp iam groups iam add-binding \
+			--group=8647ae06-ca65-467a-b72d-edba1f908fc8 \
+			--member=ef938a22-09cf-4be9-b4d0-1f4587f80f53 \
+			--role=roles/iam.group-manager
```
Comment (dadgar) [inline internal/commands/iam/groups/iam/iam.go:23]: Should this be in the example section?

**Review by dadgar — CHANGES_REQUESTED** [review]

**Review by dadgar — APPROVED** [review]

**Pull request merged**